### PR TITLE
agent meta classes in common

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
agent meta classes in common
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/CommonValue.java [hunk 1]
```diff
@@ -326,4 +333,65 @@ public class CommonValue {
...
+
+    public static final String ML_MEMORY_META_INDEX_MAPPING = "{\n"
+            + "    \"_meta\": {\"schema_version\": "
+            + ML_MEMORY_META_INDEX_SCHEMA_VERSION
+            + "    },\n"
+            + "    \"properties\": {\n"
+            + "        \"name\" : {\"type\":\"text\",\"fields\":{\"keyword\":{\"type\":\"keyword\",\"ignore_above\":256}}},\n"
+            + "        \"application_type\" : {\"type\":\"keyword\"},\n"
+            + "        \"created_time\": {\"type\": \"date\", \"format\": \"strict_date_time||epoch_millis\"},\n"
+            + "        \"last_updated_time\": {\"type\": \"date\", \"format\": \"strict_date_time||epoch_millis\"}\n"
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/CommonValue.java:380]: Seems can't match the java code https://github.com/opensearch-project/ml-commons/pull/1730/files#diff-c99fc312519dcb3fa952ea768ddcd6139fa9349669c111c186270e2869b98fdf
```
public final static String META_UPDATED_TIME_FIELD = "updated_time";
```
Suggest use the constant just like what you did for MLAgent. That can avoid mismatch.

Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/CommonValue.java:380]: Same suggest for `ML_MEMORY_MESSAGE_INDEX_MAPPING` part

Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/CommonValue.java]: makes sense, addressed.

**Review by ylwu-amzn — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/agent/LLMSpec.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+        return LLMSpec.builder()
+                .modelId(modelId)
+                .parameters(parameters)
+                .build();
+    }
+
+    public static LLMSpec fromStream(StreamInput in) throws IOException {
+        LLMSpec toolSpec = new LLMSpec(in);
+        return toolSpec;
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/agent/LLMSpec.java:100]: can we add a test here?

File: common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        }
+        if (input.readBoolean()) {
+            tools = new ArrayList<>();
+            int size = input.readInt();
+            for (int i=0; i<size; i++) {
+                tools.add(new MLToolSpec(input));
+            }
+        }
+        if (input.readBoolean()) {
+            parameters = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java:97]: Can we add more tests?

File: common/src/main/java/org/opensearch/ml/common/agent/MLMemorySpec.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        return MLMemorySpec.builder()
+                .type(type)
+                .sessionId(sessionId)
+                .windowSize(windowSize)
+                .build();
+    }
+
+    public static MLMemorySpec fromStream(StreamInput in) throws IOException {
+        MLMemorySpec toolSpec = new MLMemorySpec(in);
+        return toolSpec;
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/agent/MLMemorySpec.java:106]: add test?

Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/agent/MLMemorySpec.java:106]: added

File: common/src/main/java/org/opensearch/ml/common/agent/LLMSpec.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.utils.StringUtils.getParameterMap;
+
+
+@Getter
+public class LLMSpec implements ToXContentObject {
```
[the remark this answers is not in the document]
Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/agent/LLMSpec.java:24]: specification for LLM

File: common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java (hunk 3 above)
Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java:97]: added.

**Review by dhrubo-os — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**